cdc/owner/scheduler.go

```diff
@@ -0,0 +1,382 @@
+// Copyright 2021 PingCAP, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package owner
+
+import (
+	"math"
+
+	"github.com/pingcap/errors"
+	"github.com/pingcap/failpoint"
+	"github.com/pingcap/log"
+	"github.com/pingcap/ticdc/cdc/model"
+	cerror "github.com/pingcap/ticdc/pkg/errors"
+	"go.uber.org/zap"
+)
+
+type schedulerJobType string
+
+const (
+	schedulerJobTypeAddTable    schedulerJobType = "ADD"
+	schedulerJobTypeRemoveTable schedulerJobType = "REMOVE"
+)
+
+type schedulerJob struct {
+	Tp      schedulerJobType
+	TableID model.TableID
+	// if the operation is a delete operation, boundaryTs is checkpoint ts
+	// if the operation is a add operation, boundaryTs is start ts
+	BoundaryTs    uint64
+	TargetCapture model.CaptureID
+}
+
+type moveTableJob struct {
+	tableID model.TableID
+	target  model.CaptureID
+}
+
+type scheduler struct {
+	state         *model.ChangefeedReactorState
+	currentTables []model.TableID
+	captures      map[model.CaptureID]*model.CaptureInfo
+
+	moveTableTargets      map[model.TableID]model.CaptureID
+	moveTableJobQueue     []*moveTableJob
```

Comment:
Does these two used in manually move table, could you please add comments about how they are working